src/main/java/com/google/devtools/build/lib/remote/RemoteExecutionCache.java

```diff
@@ -57,20 +57,40 @@ private void uploadMissing(Map<Digest, Path> files, Map<Digest, ByteString> blob
       uploads.add(cacheProtocol.uploadBlob(entry.getKey(), entry.getValue()));
     }
 
-    try {
-      for (ListenableFuture<Void> upload : uploads) {
+    InterruptedException interruptedException = null;
+
+    BulkTransferException bulkTransferException = null;
+    boolean interrupted = Thread.currentThread().isInterrupted();
+    for (ListenableFuture<Void> upload : uploads) {
+      try {
         upload.get();
+      } catch (ExecutionException e) {
+        if (bulkTransferException != null) {
+          bulkTransferException = new BulkTransferException();
+        }
+        Throwable cause = e.getCause();
+        IOException ioEx;
+        if (cause instanceof IOException) {
+          ioEx = new IOException(cause);
+        } else {
+          ioEx = (IOException) cause;
+        }
```

Comment:
```suggestion
        IOException ioEx = cause instanceof IOException ? (IOException) cause : new IOException(cause);
```

Reply:
There's a better solution to this that looks more like what's available in RemoteCache.java, i'll refactor that and replace this.